### PR TITLE
Archipelago legs

### PR DESCRIPTION
**Pull request by ScipioWright**
hahahaha why did I make this

### CODE REVIEW

File: data/archipelago/scripts/patches/extend_forge_item_convert.lua
Comment (ScipioWright) [inline data/archipelago/scripts/patches/extend_forge_item_convert.lua:1]: Makes it so the leggy mutation doesn't just punch this cool skater boi right in its face

File: data/archipelago/scripts/patches/ap_extend_perk_list.lua [hunk 1]
```diff
@@ -14,6 +14,111 @@ local function ap_extend_perk_list()
...
+					component_readwrite(EntityGetFirstComponent(v, "IKLimbComponent"), { length = 50 }, function(comp)
+						comp.length = comp.length * 1.5
+					end)
+				end
+			end
+		end
+
+		local platformingcomponents = EntityGetComponent( entity_who_picked, "CharacterPlatformingComponent" )
+		if( platformingcomponents ~= nil ) then
+			for i,component in ipairs(platformingcomponents) do
```
Comment (heinermann) [inline data/archipelago/scripts/patches/ap_extend_perk_list.lua:88]:
```suggestion
			for _, component in ipairs(platformingcomponents) do
```

File: data/archipelago/scripts/patches/ap_extend_perk_list.lua [hunk 2]
```diff
@@ -14,6 +14,111 @@ local function ap_extend_perk_list()
...
+
+				ComponentSetMetaCustom( component, "run_velocity", run_speed )
+				ComponentSetMetaCustom( component, "velocity_min_x", vel_x_min )
+				ComponentSetMetaCustom( component, "velocity_max_x", vel_x_max )
+			end
+		end
+
+		perk_pickup_event("LUKKI")
+
+		if ( pickup_count <= 2 ) then
```
Comment (heinermann) [inline data/archipelago/scripts/patches/ap_extend_perk_list.lua:103]:
```suggestion
		if pickup_count <= 2 then
```

File: data/archipelago/scripts/patches/ap_extend_perk_list.lua [hunk 3]
```diff
@@ -14,6 +14,111 @@ local function ap_extend_perk_list()
...
+			add_lukkiness_level(entity_who_picked)
+		end
+	end,
+	func_remove = function( entity_who_picked )
+		reset_perk_pickup_event("LUKKI")
+		GameRemoveFlagRun( "AP_ATTACK_FOOT_CLIMBER" )
+		local platformingcomponents = EntityGetComponent( entity_who_picked, "CharacterPlatformingComponent" )
+		if( platformingcomponents ~= nil ) then
+			for i,component in ipairs(platformingcomponents) do
```
Comment (heinermann) [inline data/archipelago/scripts/patches/ap_extend_perk_list.lua:112]:
```suggestion
			for _, component in ipairs(platformingcomponents) do
```

File: data/archipelago/translations.csv [hunk 4]
```diff
@@ -39,3 +39,5 @@
 "ap_trapname10","The Best Item Ever",,,,,,,,,,,,,
 "ap_pedestal_error_name","This is an error.",,,,,,,,,,,,,
 "ap_pedestal_error_description","Starting a new run should fix this.",,,,,,,,,,,,,
+"perk_ap_leggy_feet","AP Leggy Feet",,,,,,,,,,,,,
+"perkdesc_ap_leggy_feet","You grow disturbing looking, colorful limbs that fight for you.",,,,,,,,,,,,,
```
Comment (heinermann) [inline data/archipelago/translations.csv:43]:
```suggestion
"perkdesc_ap_leggy_feet","You grow disturbing looking, colourful limbs that fight for you.",,,,,,,,,,,,,
```

The game uses UK english spelling for everything else so I figure we can be consistent.

**Review by heinermann — COMMENTED** [review]
Just a bunch of nitpicking, I don't quite understand the details on the leggy shenanigans and didn't notice anything that stood out, so if it works it works 🤷🏻‍♂️ .

**Pull request merged**